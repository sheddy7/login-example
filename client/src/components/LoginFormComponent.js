import React from 'react';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Paper from 'material-ui/Paper';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';

import '../assets/LoginForm.css';

const paperStyle = {
  height: 350,
  width: 300,
  margin: 20,
  textAlign: 'center',
  display: 'inline-block',
};

const buttonStyle = {
  margin: 30
}

const LoginFormComponent = props => {

  const {
    email,
    password,
    text,
    error,
    handleSubmit,
    handleFieldChange
  } = props;

  const testClasses = error ? 'message error' : 'message';

  return (
    <form name="loginForm" onSubmit={handleSubmit}>
      <MuiThemeProvider>
        <Paper style={paperStyle} zDepth={1}>

            <TextField
              type="email"
              name="email"
              floatingLabelText="email address"
              onChange={handleFieldChange}
              value={email}
            />


            <TextField
              type="password"
              name="password"
              floatingLabelText="password"
              onChange={handleFieldChange}
              value={password}
            />

          <RaisedButton
            style={buttonStyle}
            primary={true}
            label="Login"
            labelPosition="before"
            containerElement="label"
          >
            <input id="submitButton" type="submit" value="Login" />
          </RaisedButton>

          <div className={testClasses}>
            {text}
          </div>

        </Paper>
      </MuiThemeProvider>
    </form>
  );
};

export default LoginFormComponent;
