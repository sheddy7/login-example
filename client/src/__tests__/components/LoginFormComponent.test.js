import React from 'react';
import { mount, shallow } from 'enzyme';

import LoginFormComponent from '../../components/LoginFormComponent';

describe('<LoginFormComponent />', () => {
  it('renders without crashing', () => {
    const component = shallow(<LoginFormComponent/>);

    expect(component).toMatchSnapshot();
  });

  it('Submit works', () => {
    const testValues = {
        handleSubmit: jest.fn(),
    };

    const component = mount(
        <LoginFormComponent {...testValues} />
    );

    component.find('#submitButton').simulate('submit');
    expect(testValues.handleSubmit).toHaveBeenCalledTimes(1);
  });

  it('email, password props are shown in text fields', () => {
    const testValues = {
        email: 'test@test.com',
        password: 'test'
    };

    const component = mount(
        <LoginFormComponent {...testValues} />
    );

    const emailInput = component.find('input[name="email"]'),
      passwordInput = component.find('input[name="password"]');

    expect(emailInput).toHaveLength(1);
    expect((emailInput).props().value).toEqual(testValues.email);
    expect(passwordInput).toHaveLength(1);
    expect((passwordInput).props().value).toEqual(testValues.password);
  });

  it('passing text renders as a message without error class', () => {
    const testValues = {
      text: 'this is some text',
      error: false
    };

    const component = shallow(
      <LoginFormComponent {...testValues} />
    );

    const messageDiv = component.find('div.message');

    expect(messageDiv).toHaveLength(1);
    expect((messageDiv).text()).toEqual(testValues.text);
    expect((messageDiv).hasClass('error')).toEqual(false);
  });

  it('passing error renders as a message with error class', () => {
    const testValues = {
      text: 'something went wrong',
      error: true
    };

    const component = shallow(
      <LoginFormComponent {...testValues} />
    );

    const messageDiv = component.find('div.message');

    expect(messageDiv).toHaveLength(1);
    expect((messageDiv).text()).toEqual(testValues.text);
    expect((messageDiv).hasClass('error')).toEqual(true);
  });

  it('Updating text calls handleFieldChange', () => {
    const testValues = {
        handleFieldChange: jest.fn(),
    };

    const component = mount(
        <LoginFormComponent {...testValues} />
    );

    const emailInput = component.find('input[name="email"]'),
      passwordInput = component.find('input[name="password"]');

    emailInput.simulate('change', { target: { value: 'test@test.com' } });
    expect(testValues.handleFieldChange).toHaveBeenCalledTimes(1);

    passwordInput.simulate('change', { target: { value: 'password' } });
    expect(testValues.handleFieldChange).toHaveBeenCalledTimes(2);
  });

});
