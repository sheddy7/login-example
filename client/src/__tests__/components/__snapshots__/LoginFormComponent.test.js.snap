// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginFormComponent /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginFormComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <MuiThemeProvider>
        <Paper
                circle={false}
                rounded={true}
                style={
                        Object {
                                "display": "inline-block",
                                "height": 350,
                                "margin": 20,
                                "textAlign": "center",
                                "width": 300,
                              }
                }
                transitionEnabled={true}
                zDepth={1}
        >
                <TextField
                        disabled={false}
                        floatingLabelFixed={false}
                        floatingLabelText="email address"
                        fullWidth={false}
                        multiLine={false}
                        name="email"
                        onChange={undefined}
                        rows={1}
                        type="email"
                        underlineShow={true}
                        value={undefined}
                />
                <TextField
                        disabled={false}
                        floatingLabelFixed={false}
                        floatingLabelText="password"
                        fullWidth={false}
                        multiLine={false}
                        name="password"
                        onChange={undefined}
                        rows={1}
                        type="password"
                        underlineShow={true}
                        value={undefined}
                />
                <RaisedButton
                        containerElement="label"
                        disabled={false}
                        fullWidth={false}
                        label="Login"
                        labelPosition="before"
                        primary={true}
                        secondary={false}
                        style={
                                Object {
                                        "margin": 30,
                                      }
                        }
                >
                        <input
                                id="submitButton"
                                type="submit"
                                value="Login"
                        />
                </RaisedButton>
                <div
                        className="message"
                />
        </Paper>
</MuiThemeProvider>,
      "name": "loginForm",
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Paper
          circle={false}
          rounded={true}
          style={
                    Object {
                              "display": "inline-block",
                              "height": 350,
                              "margin": 20,
                              "textAlign": "center",
                              "width": 300,
                            }
          }
          transitionEnabled={true}
          zDepth={1}
>
          <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    floatingLabelText="email address"
                    fullWidth={false}
                    multiLine={false}
                    name="email"
                    onChange={undefined}
                    rows={1}
                    type="email"
                    underlineShow={true}
                    value={undefined}
          />
          <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    floatingLabelText="password"
                    fullWidth={false}
                    multiLine={false}
                    name="password"
                    onChange={undefined}
                    rows={1}
                    type="password"
                    underlineShow={true}
                    value={undefined}
          />
          <RaisedButton
                    containerElement="label"
                    disabled={false}
                    fullWidth={false}
                    label="Login"
                    labelPosition="before"
                    primary={true}
                    secondary={false}
                    style={
                              Object {
                                        "margin": 30,
                                      }
                    }
          >
                    <input
                              id="submitButton"
                              type="submit"
                              value="Login"
                    />
          </RaisedButton>
          <div
                    className="message"
          />
</Paper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              floatingLabelText="email address"
              fullWidth={false}
              multiLine={false}
              name="email"
              onChange={undefined}
              rows={1}
              type="email"
              underlineShow={true}
              value={undefined}
/>,
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              floatingLabelText="password"
              fullWidth={false}
              multiLine={false}
              name="password"
              onChange={undefined}
              rows={1}
              type="password"
              underlineShow={true}
              value={undefined}
/>,
            <RaisedButton
              containerElement="label"
              disabled={false}
              fullWidth={false}
              label="Login"
              labelPosition="before"
              primary={true}
              secondary={false}
              style={
                            Object {
                                          "margin": 30,
                                        }
              }
>
              <input
                            id="submitButton"
                            type="submit"
                            value="Login"
              />
</RaisedButton>,
            <div
              className="message"
/>,
          ],
          "circle": false,
          "rounded": true,
          "style": Object {
            "display": "inline-block",
            "height": 350,
            "margin": 20,
            "textAlign": "center",
            "width": 300,
          },
          "transitionEnabled": true,
          "zDepth": 1,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "floatingLabelFixed": false,
              "floatingLabelText": "email address",
              "fullWidth": false,
              "multiLine": false,
              "name": "email",
              "onChange": undefined,
              "rows": 1,
              "type": "email",
              "underlineShow": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "floatingLabelFixed": false,
              "floatingLabelText": "password",
              "fullWidth": false,
              "multiLine": false,
              "name": "password",
              "onChange": undefined,
              "rows": 1,
              "type": "password",
              "underlineShow": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <input
                id="submitButton"
                type="submit"
                value="Login"
/>,
              "containerElement": "label",
              "disabled": false,
              "fullWidth": false,
              "label": "Login",
              "labelPosition": "before",
              "primary": true,
              "secondary": false,
              "style": Object {
                "margin": 30,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "submitButton",
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "message",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <MuiThemeProvider>
          <Paper
                    circle={false}
                    rounded={true}
                    style={
                              Object {
                                        "display": "inline-block",
                                        "height": 350,
                                        "margin": 20,
                                        "textAlign": "center",
                                        "width": 300,
                                      }
                    }
                    transitionEnabled={true}
                    zDepth={1}
          >
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              floatingLabelText="email address"
                              fullWidth={false}
                              multiLine={false}
                              name="email"
                              onChange={undefined}
                              rows={1}
                              type="email"
                              underlineShow={true}
                              value={undefined}
                    />
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              floatingLabelText="password"
                              fullWidth={false}
                              multiLine={false}
                              name="password"
                              onChange={undefined}
                              rows={1}
                              type="password"
                              underlineShow={true}
                              value={undefined}
                    />
                    <RaisedButton
                              containerElement="label"
                              disabled={false}
                              fullWidth={false}
                              label="Login"
                              labelPosition="before"
                              primary={true}
                              secondary={false}
                              style={
                                        Object {
                                                  "margin": 30,
                                                }
                              }
                    >
                              <input
                                        id="submitButton"
                                        type="submit"
                                        value="Login"
                              />
                    </RaisedButton>
                    <div
                              className="message"
                    />
          </Paper>
</MuiThemeProvider>,
        "name": "loginForm",
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Paper
            circle={false}
            rounded={true}
            style={
                        Object {
                                    "display": "inline-block",
                                    "height": 350,
                                    "margin": 20,
                                    "textAlign": "center",
                                    "width": 300,
                                  }
            }
            transitionEnabled={true}
            zDepth={1}
>
            <TextField
                        disabled={false}
                        floatingLabelFixed={false}
                        floatingLabelText="email address"
                        fullWidth={false}
                        multiLine={false}
                        name="email"
                        onChange={undefined}
                        rows={1}
                        type="email"
                        underlineShow={true}
                        value={undefined}
            />
            <TextField
                        disabled={false}
                        floatingLabelFixed={false}
                        floatingLabelText="password"
                        fullWidth={false}
                        multiLine={false}
                        name="password"
                        onChange={undefined}
                        rows={1}
                        type="password"
                        underlineShow={true}
                        value={undefined}
            />
            <RaisedButton
                        containerElement="label"
                        disabled={false}
                        fullWidth={false}
                        label="Login"
                        labelPosition="before"
                        primary={true}
                        secondary={false}
                        style={
                                    Object {
                                                "margin": 30,
                                              }
                        }
            >
                        <input
                                    id="submitButton"
                                    type="submit"
                                    value="Login"
                        />
            </RaisedButton>
            <div
                        className="message"
            />
</Paper>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                floatingLabelText="email address"
                fullWidth={false}
                multiLine={false}
                name="email"
                onChange={undefined}
                rows={1}
                type="email"
                underlineShow={true}
                value={undefined}
/>,
              <TextField
                disabled={false}
                floatingLabelFixed={false}
                floatingLabelText="password"
                fullWidth={false}
                multiLine={false}
                name="password"
                onChange={undefined}
                rows={1}
                type="password"
                underlineShow={true}
                value={undefined}
/>,
              <RaisedButton
                containerElement="label"
                disabled={false}
                fullWidth={false}
                label="Login"
                labelPosition="before"
                primary={true}
                secondary={false}
                style={
                                Object {
                                                "margin": 30,
                                              }
                }
>
                <input
                                id="submitButton"
                                type="submit"
                                value="Login"
                />
</RaisedButton>,
              <div
                className="message"
/>,
            ],
            "circle": false,
            "rounded": true,
            "style": Object {
              "display": "inline-block",
              "height": 350,
              "margin": 20,
              "textAlign": "center",
              "width": 300,
            },
            "transitionEnabled": true,
            "zDepth": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "floatingLabelFixed": false,
                "floatingLabelText": "email address",
                "fullWidth": false,
                "multiLine": false,
                "name": "email",
                "onChange": undefined,
                "rows": 1,
                "type": "email",
                "underlineShow": true,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "floatingLabelFixed": false,
                "floatingLabelText": "password",
                "fullWidth": false,
                "multiLine": false,
                "name": "password",
                "onChange": undefined,
                "rows": 1,
                "type": "password",
                "underlineShow": true,
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <input
                  id="submitButton"
                  type="submit"
                  value="Login"
/>,
                "containerElement": "label",
                "disabled": false,
                "fullWidth": false,
                "label": "Login",
                "labelPosition": "before",
                "primary": true,
                "secondary": false,
                "style": Object {
                  "margin": 30,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "submitButton",
                  "type": "submit",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "message",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
